core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -618,6 +624,23 @@ impl TxSender {
         acceptable_overestimation: u32,
     ) -> Result<Fee, SubmitTxError> {
         let estimation_started_at = Instant::now();
+
+        let mut connection = self
+            .0
+            .replica_connection_pool
+            .access_storage_tagged("api")
+            .await
+            .unwrap();
+        let block_args = BlockArgs::pending(&mut connection).await;
+        // If protocol version is not present, we'll use the pre-boojum one
+        let protocol_version = connection
+            .blocks_dal()
+            .get_miniblock_protocol_version_id(block_args.resolved_block_number())
+            .await
+            .unwrap()
+            .unwrap_or(ProtocolVersionId::last_pre_boojum());
```

Comment:
i think it's safe to just unwrap

Reply:
Then maybe we should remove `Option` there at all? cc @perekopskiy

Reply:
It's not safe, ENs still don't have this data

Reply:
It's not safe, ENs don't have this data. There is a plan to perform a migration but it's not a top priority